front/app/components/InitiativeMap/index.tsx

```diff
@@ -0,0 +1,249 @@
+
+// hooks
+import useInitiativesPermissions from 'hooks/useInitiativesPermissions';
+
+// auth
+import { SuccessAction } from 'containers/Authentication/SuccessActions/actions';
+import { triggerAuthenticationFlow } from 'containers/Authentication/events';
+
+// utils
+import clHistory from 'utils/cl-router/history';
+import { stringify } from 'qs';
+import eventEmitter from 'utils/eventEmitter';
+import {
+  changeCursorOnHover,
+  getClusterConfiguration,
+  getMapPinSymbol,
+} from 'components/EsriMap/utils';
+import { useSearchParams } from 'react-router-dom';
+import { removeSearchParams } from 'utils/cl-router/removeSearchParams';
+import { IInitiativeData } from 'api/initiatives/types';
+
+// style
+import styled, { useTheme } from 'styled-components';
+
+// intl
+import { useIntl } from 'utils/cl-intl';
+import messages from './messages';
+
+export interface Props {
+  list: IInitiativeData[];
+}
+
+// Custom styling for Esri map
+const StyledMapContainer = styled(Box)`
+  calcite-action-bar {
+    display: none;
+  }
+
+  #initiative-info-node {
+    position: unset;
+  }
+`;
+
+const InitiativeMap = ({ list }: Props) => {
+  const { formatMessage } = useIntl();
+  const [searchParams] = useSearchParams();
+  const isPhoneOrSmaller = useBreakpoint('phone');
+  const theme = useTheme();
+
+  const initiativePermissions = useInitiativesPermissions('posting_initiative');
+  const clickedLocationRef = useRef<GeoJSON.Point | null>(null); // Stores the clicked map location
+
+  // Check if the URL contains a selected initiative ID. Open the initiative preview if so.
+  const initiativeIdFromUrl = searchParams.get('selected_initiative_id');
+  useEffect(() => {
+    if (initiativeIdFromUrl) {
+      removeSearchParams(['selected_initiative_id']);
+      eventEmitter.emit('initiativeSelectedEvent', initiativeIdFromUrl);
+    }
+  }, [initiativeIdFromUrl]);
+
+  // Create two div elements to use for inserting React components into Esri map popup and overlay
+  // Docs: https://developers.arcgis.com/javascript/latest/custom-ui/#introduction
+  const startInitiativeButtonNode = document.createElement('div');
+  const initiativeInfoNode = document.createElement('div');
+  initiativeInfoNode.id = 'initiative-info-node';
+
+  // Loop through initatives with locations and create array of graphics
+  const initiativesWithLocation = list.filter(
+    (initiative) => initiative?.attributes?.location_point_geojson
+  );
+  const graphics = initiativesWithLocation?.map((initiative) => {
+    return new Graphic({
+      geometry: new Point({
+        longitude:
+          initiative?.attributes?.location_point_geojson?.coordinates[0],
+        latitude:
+          initiative?.attributes?.location_point_geojson?.coordinates[1],
+      }),
+      attributes: {
+        initiativeId: initiative?.id,
+      },
+    });
+  });
+
+  // Create an Esri map layer from the graphics so we can add a cluster display
+  const initiativesLayer = graphics
```

Comment:
You will probably need to memoize this anyway (see suggestions for `EsriMap` above), but these seem like pretty large operations, and it would be good to memoize them in general

Reply:
Ok noted! I'll make sure to memoize these large operations from now on :+1: Thanks! :)